### PR TITLE
Add a random selection of type annotations

### PR DESCRIPTION
**Pull request by eric-wieser**
As requested by @utensil in gh-303

### CODE REVIEW

File: galgebra/mv.py [hunk 1]
```diff
@@ -67,7 +72,7 @@ class Mv(object):
     dual_mode_lst = ['+I', 'I+', '+Iinv', 'Iinv+', '-I', 'I-', '-Iinv', 'Iinv-']
 
     @staticmethod
-    def setup(ga):
+    def setup(ga) -> Tuple['Mv', List['Mv'], 'Mv']:
```
Comment (utensil) [inline galgebra/mv.py:75]: The type of `ga` is missing?

File: galgebra/mv.py [hunk 2]
```diff
@@ -1911,7 +1916,7 @@ def correlation(u, v, dec=3):  # Compute the correlation coefficient of vectors
     return ulocal.dot(vlocal) / (ulocal.norm() * vlocal.norm()). evalf(dec)
 
 
-def cross(v1, v2):
+def cross(v1: Mv, v2: Mv) -> Mv:
```
Comment (utensil) [inline galgebra/mv.py:1919]: Why sometimes `'Mv'` and sometimes `Mv`?

Comment (eric-wieser) [reply galgebra/mv.py:1919]: Within `class Mv`, `Mv` doesn't exist yet, so can't be used directly.

File: galgebra/mv.py (hunk 1 above)
Comment (eric-wieser) [reply galgebra/mv.py:75]: Good catch,
```suggestion
    def setup(ga: 'Ga') -> Tuple['Mv', List['Mv'], 'Mv']:
```

File: galgebra/mv.py [hunk 3]
```diff
@@ -15,13 +15,18 @@
...
+from sympy.printing.str import StrPrinter as _StrPrinter
 
 from . import printer
 from . import metric
 from .printer import ZERO_STR
 from ._utils import KwargParser as _KwargParser
 from . import dop
 
+if TYPE_CHECKING:
+    from galgebra.ga import Ga
```
Comment (utensil) [inline galgebra/mv.py:28]: What purpose does this line serve?

File: galgebra/mv.py (hunk 2 above)
Comment (utensil) [reply galgebra/mv.py:1919]: Oh.......

File: galgebra/mv.py (hunk 3 above)
Comment (eric-wieser) [reply galgebra/mv.py:28]: `Ga` needs to be imported in order to use as a type annotation - but if we import it, we'd end up with a circular import. https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING is a special marker to indicate that the variable is there for `flake8` and `mypy`, but not at runtime.

File: galgebra/mv.py (hunk 2 above)
Comment (eric-wieser) [reply galgebra/mv.py:1919]: Note they all render the same way in the docs: https://galgebra--305.org.readthedocs.build/en/305/generated/galgebra.mv.html#module-galgebra.mv

Comment (utensil) [reply galgebra/mv.py:1919]: Not necessarily: https://stackoverflow.com/a/33533514

File: galgebra/mv.py (hunk 3 above)
Comment (utensil) [reply galgebra/mv.py:28]: So the fact is these function actually is relying on circular import, we need to move them out someday ( I move a few back then to break circular dependency but these seem to be the ones I missed)

File: galgebra/mv.py (hunk 2 above)
Comment (utensil) [reply galgebra/mv.py:1919]: But as long as we support Python below 3.7, using string is inevitable.

File: galgebra/mv.py (hunk 3 above)
Comment (eric-wieser) [reply galgebra/mv.py:28]: No, the function isn't relying on a circular import - just the type annotations are. That's to be expected in my opinion.

The `Ga` class _creates_ `Mv` instances, so needs a direct import of the type. However, the `Mv` class only  _uses_ `Ga` instances passed in from elsewhere, so only needs `Ga` "imported" as a documentation hint.

File: galgebra/mv.py (hunk 2 above)
Comment (eric-wieser) [reply galgebra/mv.py:1919]: Indeed, but we still support 3.5 and 3.6, so this is the preferred approach for now.

File: galgebra/mv.py (hunk 3 above)
Comment (utensil) [reply galgebra/mv.py:28]: I see

**Pull request merged**